flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java

```diff
@@ -164,6 +164,7 @@ public void triggerCheckpoint(long checkpointId) throws FlinkException {
                 .gauge(
                         MetricNames.CHECKPOINT_START_DELAY_TIME,
                         this::getAsyncCheckpointStartDelayNanos);
+        recordWriter.setMaxOverdraftBuffersPerGate(0);
```

Comment:
Could you add a unit test for this? I think it could look similar to `SourceStreamTaskTest#testTriggeringCheckpointAfterSourceThreadFinished` but vastly simplified:

1. Create a single additional `ResultPartition`
2. assert `resultPartition.getBufferPool().getMaxOverdraftBuffersPerGate() > 0`
3. Construct `SourceStreamTask` via `StreamTaskMailboxTestHarnessBuilder`
4. assert `resultPartition.getBufferPool().getMaxOverdraftBuffersPerGate() == 0`
 
?

Reply:
Thanks for your review, added.